File: JavaCourse/lab3/work/src/main/java/ua/lviv/iot/Plane.java
package ua.lviv.iot;

import javax.persistence.*;

@Entity
public class Plane {


    @Id
    @Column(name = "id")
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "load_capacity")
    private double loadCapacity;

    @Column(name = "capacity")
    private int capacity;

    @Column(name = "fuel_consumption")
    private double fuelConsumption;

    @Column(name = "flight_range")
    private double flightRange;

    public Plane() {
    }

    public Plane(final String pName, final int pCapacity,
                 final double pLoadCapacity, final double pFuelConsumption,
                 final double pFlightRange) {
        setName(pName);
        setCapacity(pCapacity);
        setLoadCapacity(pLoadCapacity);
        setFuelConsumption(pFuelConsumption);
        setFlightRange(pFlightRange);
    }

    public Plane(final String pName, final int pCapacity,
                 final double pLoadCapacity, final double pFuelConsumption,
                 final double pFlightRange, final int id) {
        setName(pName);
        setCapacity(pCapacity);
        setLoadCapacity(pLoadCapacity);
        setFuelConsumption(pFuelConsumption);
        setFlightRange(pFlightRange);
        setId(id);
    }

    protected String getHeaders() {
        return "name,type,capacity,loadCapacity,fuelConsumption,flightRange";
    }

    protected String toCSV() {
        return capacity + ", " + loadCapacity + ", " + fuelConsumption + ", " + flightRange;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getLoadCapacity() {
        return loadCapacity;
    }

    public void setLoadCapacity(double loadCapacity) {
        this.loadCapacity = loadCapacity;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }

    public double getFuelConsumption() {
        return fuelConsumption;
    }

    public void setFuelConsumption(double fuelConsumption) {
        this.fuelConsumption = fuelConsumption;
    }

    public double getFlightRange() {
        return flightRange;
    }

    public void setFlightRange(double flightRange) {
        this.flightRange = flightRange;
    }

}

File: PythonCourse/lab2/Plane.py
class Plane:
    def __init__(self, name = "", capacity = 0, load_capacity = 0, fuel_consumption = 0, flight_range = 0):
        self.name = name
        self.capacity = capacity
        self.load_capacity = load_capacity
        self.fuel_consumption = fuel_consumption
        self.flight_range = flight_range

    def to_string(self):
        return "Name: " + self.name + ", capacity: " + str(self.capacity) + ", load_capacity: " + \
               str(self.load_capacity) + ", fuel_consumption: " + str(self.fuel_consumption) + ", flight_range: " + \
               str(self.flight_range)


class Civil(Plane):
    def __init__(self, name = "", capacity = 0, load_capacity = 0, fuel_consumption = 0,
                 flight_range = 0, plane_type = "Civil"):
        self.plane_type = plane_type
        Plane.__init__(self, name, capacity, load_capacity, fuel_consumption, flight_range)

    def to_string(self):
        return Plane.to_string(self) + ", plane type: " + self.plane_type


class Military(Plane):
    def __init__(self, name = "", capacity = 0, load_capacity = 0, fuel_consumption = 0,
                 flight_range = 0, plane_type = "Military"):
        Plane.__init__(self, name, capacity, load_capacity, fuel_consumption, flight_range)
        self.plane_type = plane_type

    def to_string(self):
        return Plane.to_string(self) + ", plane type: " + self.plane_type
